### ISSUE TITLE
Bogus invariant_booleans analyzer warning

### ISSUE THREAD

**Issue by apwilson** [dart-lang/linter#1859]
This code has an analyzer warning that's bogus imo.
```dart
enum _Foo { x, y, z }

///
class Baz {
  final double _a;
  final double _b;

  ///
  Baz(this._a, this._b);
  bool get _isHorizontal => _a == _b;

  ///
  bool doSomething(_Foo foo) {
    switch (foo) {
      case _Foo.x:
        if (_isHorizontal) {
          return false;
        }
        break;
      case _Foo.y:
        if (_isHorizontal) {
          return false;
        }
        break;
      case _Foo.z:
      default:
        break;
    }
    return true;
  }
}

```

**Comment by jamesderlin** [comment]
This looks rather similar to https://github.com/dart-lang/linter/issues/720 (multiple `switch` `case`s with a repeated condition).

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
